internal/schema/schema.go

```diff
@@ -259,6 +259,9 @@ func (m *manager) SubscriberID() string {
 
 func (m *manager) OnStorageEvent(events ...storage.Event) {
 	for _, event := range events {
+		if event.IndexUnhealthy {
```

Comment:
I don't think this is necessary because we can still continue to load schemas even if the index is unhealthy. 

- If it's an update to an existing schema, we should apply it because we may never get the schema update event again because those are rare.
- If it's a new schema, there's no harm in adding it to the schema cache.